pkg/kubectl/cmd/apply_set_last_applied.go

```diff
@@ -41,22 +41,23 @@ import (
 )
 
 type SetLastAppliedOptions struct {
-	FilenameOptions  resource.FilenameOptions
-	Selector         string
-	InfoList         []*resource.Info
-	Mapper           meta.RESTMapper
-	Typer            runtime.ObjectTyper
-	Namespace        string
-	EnforceNamespace bool
-	DryRun           bool
-	ShortOutput      bool
-	CreateAnnotation bool
-	Output           string
-	Codec            runtime.Encoder
-	PatchBufferList  [][]byte
-	Factory          cmdutil.Factory
-	Out              io.Writer
-	ErrOut           io.Writer
+	FilenameOptions    resource.FilenameOptions
+	Selector           string
+	InfoList           []*resource.Info
+	Mapper             meta.RESTMapper
+	Typer              runtime.ObjectTyper
+	ApplyPatchTypeList []types.PatchType
```

Comment:
prefer a list of structs containing the patch content and patch type over parallel lists:

```
PatchBufferList    []PatchBuffer
...

type PatchBuffer struct {
	Patch     []byte
	PatchType types.PatchType
}
```